### PR TITLE
DROTH-1791_csv_importer_improvement

### PR DESCRIPTION
**Pull request by moraispedro**

### CODE REVIEW

File: UI/excel_import.html [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
-        </div>
-    </div>
-    <script type="text/javascript" src="src/utils/environment-utils.js"></script>
-    <script type="text/javascript" src="src/analytics.js"></script>
-    <script type="application/javascript">
-        Analytics.start();
-    </script>
-</body>
-</html>
+<!--<!DOCTYPE html>-->
```
Comment (oskarmalmstrom) [inline UI/excel_import.html]: to be removed

File: UI/src/csvImporter.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+  function getJobs() {
+     backend.getJobs().then(function(jobs){
+      if(!_.isEmpty(jobs)) {
+        $('.job-status').empty().html(buildJobTable(jobs));
+        rootElement.find('.job-status-link').on('click', function (event) {
+          getJob(event);
+        });
+        if(!refresh)
+          refresh = setInterval(refreshJobs, 30000);
```
Comment (oskarmalmstrom) [inline UI/src/csvImporter.js]: what should be the interval for fetching new jobs?

File: digiroad2-api-common/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Context.scala [hunk 3]
```diff
@@ -246,8 +253,8 @@ class TrafficSignCreateAssets(trafficSignManager: TrafficSignManager) extends Ac
 
 class TrafficSignExpireAssets(trafficSignService: TrafficSignService, trafficSignManager: TrafficSignManager) extends Actor {
   def receive = {
-    case x: Long => trafficSignService.getTrafficType(x) match {
-      case Some(trafficType) => trafficSignManager.deleteAssets(x, trafficType)
+    case x: Long => trafficSignService.getPersistedAssetsByIdsWithExpire(Set(x)).headOption match {
```
Comment (oskarmalmstrom) [inline digiroad2-api-common/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Context.scala]: automatic process code

File: UI/src/csvImporter.js (hunk 2 above)
Comment (moraispedro) [reply UI/src/csvImporter.js]: I would say something like 5 to 10 seconds. Seems enough to me.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/ImportLogService.scala [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+////    OracleDatabase.withDynTransaction {
+////      importLogDao.getByUser(username, Seq(logType))
+////    }
+////  }
+////}
+//
+////class ImportLogService extends ImportLogServiceOperation {
+////  override def getByUser(username: String) : Seq[ImportStatusInfo] = throw new UnsupportedOperationException("Not supported method")
+////  override val logType : String = ""
+////}
```
Comment (moraispedro) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/ImportLogService.scala]: This file is all commented. Is it needed?

File: src/main/scala/ScalatraBootstrap.scala [hunk 5]
```diff
@@ -31,8 +31,8 @@ ScalatraBootstrap extends LifeCycle {
     context.mount(new SessionApi, "/api/auth/*")
     context.mount(new UserConfigurationApi, "/api/userconfig/*")
     context.mount(new PingApi, "/api/ping/*")
-    context.mount(new MassTransitStopImportApi, "/api/import/*")
-    context.mount(new ImportDataApi, "/api/import/data/*")
+//    context.mount(new MassTransitStopImportApi, "/api/import/*")
```
Comment (marquesfm) [inline src/main/scala/ScalatraBootstrap.scala]: Commented code, if not necessary please remove it

File: digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/ImportDataApi.scala [hunk 6]
```diff
@@ -1,139 +1,143 @@
...
+    eventBus.publish("importCSVData", CsvDataImporterInfo("roadLinks", fileName, user, csvFileInputStream))
+  }
+
+  def importMaintenanceRoads(csvFileItem: FileItem): Unit = {
+    val csvFileInputStream = csvFileItem.getInputStream
+    val fileName = csvFileItem.getName
+    if (csvFileInputStream.available() == 0) halt(BadRequest("Ei valittua CSV-tiedostoa. Valitse tiedosto ja yritä uudestaan.")) else None
+
+    eventBus.publish("importCSVData", CsvDataImporterInfo(MaintenanceRoadAsset.layerName, fileName, user, csvFileInputStream))
+  }
```
Comment (MAC-CGI) [inline digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/ImportDataApi.scala]: same idea as previous

File: digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/dataimport/CsvImporterSpec.scala [hunk 7]
```diff
@@ -1,389 +1,388 @@
...
+//    val result = importer.importAssets(inputStream, roadTypeLimitations = Set(Private))
+//    result should equal(ImportResultMassTranstiStop(
+//      excludedAssets = List(ExcludedAsset(affectedRoadLinkType = "State", csvRow = csvImporter.rowToString(defaultValues ++ assetOnRoadFields)),
+//        ExcludedAsset(affectedRoadLinkType = "Municipality", csvRow = csvImporter.rowToString(defaultValues ++ assetOnStreetFields)))))
+//
+//    verify(mockMassTransitStopService, never).updateExistingById(anyLong(), anyObject(), anyObject(), anyString(), anyObject())
+//  }
+//
+//  private def csvToInputStream(csv: String): InputStream = new ByteArrayInputStream(csv.getBytes())
+//}
```
[the remark this answers is not in the document]
Comment (MAC-CGI) [reply digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/dataimport/CsvImporterSpec.scala]: Should be corrected

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/CsvDataImporter.scala [hunk 8]
```diff
@@ -0,0 +1,1109 @@
...
+  private def verifyValueCode(parameterName: String, parameterValue: String): ParsedRow = {
+    if(parameterValue.forall(_.isDigit) && TrafficSignType.applyTRValue(parameterValue.toInt).source.contains("CSVimport")){
+      (Nil, List(AssetProperty(columnName = codeValueFieldMappings(parameterName), value = parameterValue.toInt)))
+    }else{
+      (List(parameterName), Nil)
+    }
+  }
+
+  def getRoadLinkByBearing(assetBearing: Option[Int], assetValidityDirection: Option[Int], assetCoordinates: Point, roadLinks: Seq[RoadLink]): Seq[RoadLink] = {
+    val toleranceInDegrees = 25
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/CsvDataImporter.scala]: **Pay attention with that method on merge 1746 CSV_import_modify_v2**

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/CsvDataImporter.scala [hunk 9]
```diff
@@ -0,0 +1,1109 @@
...
+      else
+        None
+
+    val typeProperty = SimpleTrafficSignProperty(typePublicId, Seq(TextPropertyValue(TrafficSignType.applyTRValue(getPropertyValue(trafficSignAttributes, "trafficSignType").toString.toInt).OTHvalue.toString)))
+
+    Set(Some(typeProperty), valueProperty, additionalProperty).flatten
+  }
+
+  def recalculateBearing(bearing: Option[Int]): (Option[Int], Option[Int]) = {
+    bearing match {
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/CsvDataImporter.scala]: **Pay attention with that method on merge 1746 CSV_import_modify_v2**

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/CsvDataImporter.scala [hunk 10]
```diff
@@ -0,0 +1,1109 @@
...
+            Math.abs(assetBearing - 180)
+        } else assetBearing
+
+        (Some(readjustedBearing), Some(validityDirection))
+      case _ =>
+        (None, None)
+    }
+  }
+
+  def createTrafficSigns(trafficSignAttributes: Seq[CsvAssetRowAndRoadLink], user: User, result: ImportResultTrafficSign ): ImportResultTrafficSign = {
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/CsvDataImporter.scala]: **Pay attention with that method on merge 1746 CSV_import_modify_v2**

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PointAssetOperations.scala [hunk 11]
```diff
@@ -310,12 +310,24 @@ trait PointAssetOperations {
     }
   }
 
+  def getPersistedAssetsByIdsWithExpire(ids: Set[Long]): Seq[PersistedAsset] = {
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PointAssetOperations.scala]: check if that method is in use

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PointAssetOperations.scala [hunk 12]
```diff
@@ -310,12 +310,24 @@ trait PointAssetOperations {
...
+    }
+  }
+
   def getPersistedAssetsByIdsWithoutTransaction(ids: Set[Long]): Seq[PersistedAsset] = {
       val idsStr = ids.toSeq.mkString(",")
       val filter = s"where a.asset_type_id = $typeId and a.id in ($idsStr)"
       fetchPointAssets(withFilter(filter))
   }
 
+  def getPersistedAssetsByIdsWithExpireWithoutTransaction(ids: Set[Long]): Seq[PersistedAsset] = {
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PointAssetOperations.scala]: check if that method is in use

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/pointasset/OracleTrafficSignDao.scala [hunk 13]
```diff
@@ -99,7 +99,7 @@ object OracleTrafficSignDao {
     sql"""select distinct ev.id from PROPERTY p
                 join ENUMERATED_VALUE ev on ev.property_id = p.id
                 join SINGLE_CHOICE_VALUE sc on sc.ENUMERATED_VALUE_ID = ev.id
-                where p.ASSET_TYPE_ID = 300 and p.PUBLIC_ID = 'trafficSigns_type' and ev.VALUE in (#${values.mkString(",")}) """.as[Long].list
+                where p.ASSET_TYPE_ID = 300 and p.PUBLIC_ID = 'trafficSigns_type' and ev.VALUE in logIds """.as[Long].list
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/pointasset/OracleTrafficSignDao.scala]: this change is incorrect, **revert**

File: src/main/scala/ScalatraBootstrap.scala (hunk 5 above)
Comment (MAC-CGI) [inline src/main/scala/ScalatraBootstrap.scala]: this commented line should be deleted

File: UI/src/csvImporter.js [hunk 14]
```diff
@@ -0,0 +1,216 @@
+var municipalities;
+var backend =  new CsvBackend();
+var refresh;
+
+$(function() {
```
[the remark this answers is not in the document]
Comment (oskarmalmstrom) [reply UI/src/csvImporter.js]: where would it be better to store it?

**Pull request merged**